core/chains/evm/logpoller/log_poller.go

```diff
@@ -447,16 +481,36 @@ func (lp *logPoller) findBlockAfterLCA(ctx context.Context, current *types.Heade
 		}
 	}
 	lp.lggr.Criticalw("Reorg greater than finality depth detected", "max reorg depth", lp.finalityDepth-1)
-	return nil, errors.New("reorg greater than finality depth")
+	return nil, errors.New("Reorg greater than finality depth")
+}
+
+func (lp *logPoller) assertInFilter(eventSigs []common.Hash, addresses []common.Address) error {
+	for _, eventSig := range eventSigs {
+		if _, ok := lp.eventSigs[eventSig]; !ok {
+			return errors.Errorf("eventSig %x not registered", eventSig)
+		}
+	}
+	for _, addr := range addresses {
+		if _, ok := lp.addresses[addr]; !ok {
+			return errors.Errorf("address %x not registered", addr)
+		}
+	}
+	return nil
 }
 
 // Logs returns logs matching topics and address (exactly) in the given block range,
 // which are canonical at time of query.
 func (lp *logPoller) Logs(start, end int64, eventSig common.Hash, address common.Address, qopts ...pg.QOpt) ([]Log, error) {
+	if err := lp.assertInFilter([]common.Hash{eventSig}, []common.Address{address}); err != nil {
```

Comment:
This would not allow querying logs that have had their sig/address removed after being added to the DB. I don't see a need to allow this but I'm curious why we explicitly want to disallow this

Reply:
yeah I'm not sure of a use case for querying logs for which you haven't registered (yet), so seems safer to directly return an error when querying for unregistered events vs just a silent empty log set

Reply:
I was wondering if there is a case for looking at logs after unsubscribing. So the logs will be there, from a previous job for example, but the filter is no longer running.

Potential use case might be checking how far back the backfill should start when adding a new job after (long) failure, as the last correct log would be the lower bound.

Reply:
In that case, it'd still add its filter when the job is added